sunpy/visualization/animator/line.py

```diff
@@ -95,11 +95,16 @@ def __init__(self, data, plot_axis_index=-1, axis_ranges=None, ylabel=None, xlab
             axis_ranges = None
         if axis_ranges is None or axis_ranges[self.plot_axis_index] is None:
             self.xdata = np.arange(data.shape[self.plot_axis_index])
+        # Else derive the xdata as pixel centers from the pixel edges supplied by the user in axis_ranges[plot_axis_index]
         else:
-            # Else derive the xdata as the centers of the pixel/bin edges
-            # supplied by the user for the plotted axis.
-            self.xdata = edges_to_centers_nd(np.asarray(axis_ranges[self.plot_axis_index]),
-                                             plot_axis_index)
+            # If the shape of the array is a 1D array, get the centers about axis=0
+            if np.asarray(axis_ranges[self.plot_axis_index]).ndim == 1:
+                self.xdata = edges_to_centers_nd(np.asarray(axis_ranges[self.plot_axis_index]), 0)
+
+            # Else derive the xdata as pixel centers from the pixel edges supplied by the user in axis_ranges[plot_axis_index]
+            # about axis=plot_axis_index
```

Comment:
Probably clearer to say ```along``` here instead of ```about```.

Reply:
Sure